src/System.Linq.Parallel/tests/Helpers/Enumerables.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Test
+{
+    internal class ThrowOnEnumeration<T> : IEnumerable<T>
```

Comment:
... I'm liking LINQ, but tend to still think too much in creating my own enumerables.  Thanks, though: stealing.